File: backend/app/schemas/analytics.py
from pydantic import BaseModel


class KPIs(BaseModel):
    total_today: int
    total_yesterday: int
    total_week: int
    total_month: int
    satisfaction_rate: float         # esta semana: % rating >= 4
    avg_response_time_ms: float      # esta semana: avg response time
    prev_week_satisfaction_rate: float
    prev_week_avg_response_time_ms: float


class DailyVolume(BaseModel):
    date: str
    count: int


class HourlyVolume(BaseModel):
    hour: int   # 0-23
    count: int


class RatingDistribution(BaseModel):
    rating: int
    count: int
    percentage: float


class ChannelDistribution(BaseModel):
    channel: str
    count: int
    percentage: float


class PromptStats(BaseModel):
    prompt_name: str
    avg_rating: float
    conversation_count: int


class AnalyticsSummary(BaseModel):
    kpis: KPIs
    daily_volume: list[DailyVolume]
    hourly_volume: list[HourlyVolume]
    rating_distribution: list[RatingDistribution]
    channel_distribution: list[ChannelDistribution]
    worst_prompts: list[PromptStats]

File: frontend/src/api/analytics.ts
import api from "./client";

export interface KPIs {
  total_today: number;
  total_week: number;
  total_month: number;
  satisfaction_rate: number;
  avg_response_time_ms: number;
}

export interface DailyVolume {
  date: string;
  count: number;
}

export interface RatingDistribution {
  rating: number;
  count: number;
  percentage: number;
}

export interface ChannelDistribution {
  channel: string;
  count: number;
  percentage: number;
}

export interface PromptStats {
  prompt_name: string;
  avg_rating: number;
  conversation_count: number;
}

export interface AnalyticsSummary {
  kpis: KPIs;
  daily_volume: DailyVolume[];
  rating_distribution: RatingDistribution[];
  channel_distribution: ChannelDistribution[];
  worst_prompts: PromptStats[];
}

export async function getAnalytics(): Promise<AnalyticsSummary> {
  const { data } = await api.get<AnalyticsSummary>("/analytics");
  return data;
}
